h5py/tests/test_dataset.py

```diff
@@ -1611,3 +1611,13 @@ def test_setitem_fancy_indexing(writable_file):
     arr = writable_file.create_dataset('data', (5, 1000, 2), dtype=np.uint8)
     block = np.random.randint(255, size=(5, 3, 2))
     arr[:, [0, 2, 4], ...] = block
+
+
+def test_allow_unknown_filter(writable_file):
+    # apparently 256-511 are reserved for testing purposes
+    fake_filter_id = 256
+    ds = writable_file.create_dataset(
+        'data', shape=(10, 10), dtype=np.uint8, compression=fake_filter_id,
+        allow_unknown_filter=True
```

Comment:
Hi, I guess the CI failed in here because you allow the unknown filter, but it can't set filter correctly at here: https://github.com/h5py/h5py/blob/6f4c578f78321b857da31eee0ce8d9b1ba291888/h5py/_hl/filters.py#L275

Reply:
Interestingly it does pass on a number of systems, I wonder if it's related to the HDF5 version?

Reply:
Looks like those failing are HDF5 1.10.1 or less, I'm going to try some local builds of early 1.10, and see if I find where the fix is.

Reply:
1.10.2 didn't seem to build for me, if 1.10.3 works, then I might bump the HDF5 versions on appveyor to 1.10.3.